tests/test_integrated_channels/test_blackboard/test_client.py

```diff
@@ -685,3 +685,125 @@ def test_content_child_customization_failure(self):
+                responses.GET,
+                client.generate_gradebook_url(self.blackboard_course_id),
+                json=blackboard_page_one_response
+            )
+            rsps.add(
+                responses.GET,
+                client.generate_gradebook_url(self.blackboard_course_id),
+                json=blackboard_page_two_response
+            )
+
+            # find the grade column on the second page of results
+            column_id = client._get_or_create_integrated_grade_column(  # pylint: disable=protected-access
+                self.blackboard_course_id,
+                self.blackboard_grade_column_name,
+                self.course_id
+            )
+
+        assert column_id == self.blackboard_grade_column_id
+
+    def test_pagination_properly_handles_column_not_found(self):
+        """
+        Test that _get_or_create_integrated_grade_column will properly exhaust all pages of results before determining
+        that a gradebook column does not exist.
+        """
+        client = self._create_new_mock_client()
+        client._create_session()  # pylint: disable=protected-access
+
+        blackboard_page_one_response = {
+            'results': [{
+                'id': 'bad ID',
+                'name': 'Weighted Total',
+                'description': 'description',
+                'score': {'possible': 0.0},
+                'availability': {'available': 'Yes'},
+                'grading': {'type': 'Calculated'}
+            }],
+            'paging': {
+                'nextPage': '/learn/api/public/v1/courses/{}/gradebook/columns?offset=1'.format(
+                    self.blackboard_course_id
+                )
+            },
+        }
+        blackboard_page_two_response = {
+            'results': [{
+                'id': 'bad ID 2',
+                'name': 'Final Grade',
+                'score': {'possible': 0.0},
+                'availability': {'available': 'Yes'},
+                'grading': {'type': 'Calculated'}
+            }],
+        }
+
+        with responses.RequestsMock() as rsps:
+            # Requests mock will iteratively return registered responses on subsequent calls
+            rsps.add(
+                responses.GET,
+                client.generate_gradebook_url(self.blackboard_course_id),
+                json=blackboard_page_one_response
+            )
+            rsps.add(
+                responses.GET,
+                client.generate_gradebook_url(self.blackboard_course_id),
+                json=blackboard_page_two_response
+            )
+            rsps.add(
+                responses.POST,
+                client.generate_create_grade_column_url(self.blackboard_course_id),
+                json={'id': self.blackboard_grade_column_id}
+            )
+
+            # traverse over all pages, post a new column if column isn't found
+            column_id = client._get_or_create_integrated_grade_column(  # pylint: disable=protected-access
```

Comment:
Can you add some kind of test here that we call the pages the correct number of times? I don't see a test here that actually confirms that we make that call to page 2?

Reply:
`responses.RequestsMock` by default asserts that all mock created within the context must be executed before leaving. If we didn't make the call to page 2 then the test would fail with `AssertionError: Not all requests have been executed [('GET', 'https://base.url/learn/api/public/v1/courses/723/gradebook/columns')]`

Reply:
Sweet, TIL!